gnucash/gnome/dialog-fincalc.c

```diff
@@ -153,12 +157,12 @@ fi_to_gui(FinCalcDialog *fcd)
 
     pmt = double_to_gnc_numeric (fcd->financial_info.pmt, 100000, GNC_HOW_RND_ROUND_HALF_UP);
 
-    commodity = gnc_default_currency ();
+    precision = gtk_spin_button_get_value_as_int(GTK_SPIN_BUTTON(fcd->precision));
+    scu = (int64_t) pow(10, precision);
 
-    total = gnc_numeric_mul (npp, pmt, gnc_commodity_get_fraction (commodity),
-                             GNC_HOW_RND_ROUND_HALF_UP);
+    total = gnc_numeric_mul (npp, pmt, scu, GNC_HOW_RND_ROUND_HALF_UP);
```

Comment:
gnc-numeric has GNC_HOW_ROUND_SIGFIGS for this use-case. But since you're telling `xaccSPrintAmount the number of places to print, why round at all?